src/main/java/qupath/bioimageio/spec/BioimageIoSpec.java

```diff
@@ -1511,5 +1546,250 @@ public Processing.ProcessingMode deserialize(JsonElement json, Type typeOfT, Jso
+			super(id, description);
+			int s = 1;
+			if (size != null) {
+				s = size.getAsInt();
+			}
+			this.size = s;
+		}
+	}
+
+	public static class ChannelAxis extends AxisBase {
+		private List<String> channel_names;
+
+		public ChannelAxis(JsonElement id, JsonElement description, List<String> channel_names) {
+			super(id, description);
+			this.channel_names = channel_names;
+		}
+
+		public int size() {
+			return channel_names.size();
+		}
+	}
+
+	public static class IndexAxisBase extends AxisBase {
+		private double scale = 1.0;
+		private String unit = null;
+
+		public IndexAxisBase(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class IndexInputAxis extends IndexAxisBase {
+		private boolean concatenable = false;
+
+		public IndexInputAxis(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class IndexOutputAxis extends IndexAxisBase {
+		private Size size;
+
+		public IndexOutputAxis(JsonElement id, JsonElement description) {
+			super(id, description);
+		}
+	}
+
+	public static class Size {
+		// todo: handle ints or SizeReference of DataDependentSize...???
+	}
+
+	public static class DataDependentSize extends Size {
+		private int min = 1;
+		private int max = Integer.MAX_VALUE;
+	}
+
+	public static class SizeReference extends Size {
+		private Axis referenceAxis; // todo: what class here?
+		private int offset;
+	}
+
+	public static class TimeAxisBase extends AxisBase {
+		private final String type = "time";
+		private final TimeUnit unit;
+		private final double scale;
+
+		TimeAxisBase(JsonElement id, JsonElement description, TimeUnit unit, double scale) {
+			super(id, description);
+			this.unit = unit;
+			this.scale = scale;
+		}
+	}
+
+	public static enum TimeUnit {
+		ATTOSECOND,
+		CENTISECOND,
+		DAY,
+		DECISECOND,
+		EXASECOND,
+		FEMTOSECOND,
+		GIGASECOND,
+		HECTOSECOND,
+		HOUR,
+		KILOSECOND,
+		MEGASECOND,
+		MICROSECOND,
+		MILLISECOND,
+		MINUTE,
+		NANOSECOND,
+		PETASECOND,
+		PICOSECOND,
+		SECOND,
+		TERASECOND,
+		YOCTOSECOND,
+		YOTTASECOND,
+		ZEPTOSECOND,
+		ZETTASECOND;
+	}
+
+	public static enum SpaceUnit {
+		ATTOMETER,
+		ANGSTROM,
+		CENTIMETER,
+		DECIMETER,
+		EXAMETER,
+		FEMTOMETER,
+		FOOT,
+		GIGAMETER,
+		HECTOMETER,
+		INCH,
+		KILOMETER,
+		MEGAMETER,
+		METER,
+		MICROMETER,
+		MILE,
+		MILLIMETER,
+		NANOMETER,
+		PARSEC,
+		PETAMETER,
+		PICOMETER,
+		TERAMETER,
+		YARD,
+		YOCTOMETER,
+		YOTTAMETER,
+		ZEPTOMETER,
+		ZETTAMETER;
+	}
+
+	public enum BioImageIoVersion {
+		VERSION_0_4("0.4"),
+		VERSION_0_5("0.5");
+		private final String version;
+
+		BioImageIoVersion(String s) {
+			this.version = s;
+		}
+	}
+
+	public static class SpaceAxisBase extends AxisBase {
+		private SpaceUnit unit;
+		private double scale = 1.0;
+
+		public SpaceAxisBase(JsonElement id, JsonElement description, String unit, double scale) {
```

Comment:
Do these constructors need to be public?
Ideally we wouldn't leak Gson anywhere into the API - since the files aren't even using JSON. We're just using it currently for convenience, but would like to be able to move away from it.